misk/src/main/kotlin/misk/healthchecks/HealthCheck.kt

```diff
@@ -0,0 +1,21 @@
+package misk.healthchecks
+
+import misk.web.actions.ReadinessCheckAction
+
+/**
+ * Allows users to define custom health checks. An app with a failing HealthCheck will fail the
+ * readiness check in [ReadinessCheckAction], indicating that the app should not accept traffic.
+ */
+interface HealthCheck {
+    /**
+     * Computes whether a component of an application is healthy. For example, an implementing class
+     * can check database connectivity. As this method may involve a network call, callers should
+     * cache the response appropriately.
+     */
+    fun status(): HealthStatus
+}
+
+data class HealthStatus(
+    val isHealthy: Boolean,
+    val messages: List<String>
```

Comment:
It'd be nice to have companion functions for creating healthy and unhealthy status, so you don't have to remember whether the boolean means healthy or not when you see it written in code. e.g.

````
companion object {
   fun unhealthy(vararg messages: String) = HealthStatus(false, listOf(messages))
   fun healthy() = HealthStatus(true, listOf())
}
````

Reply:
Added